src/runtime/HalideBuffer.h

```diff
@@ -1938,89 +1942,72 @@ class Buffer {
     // (all of different types), advance the pointers using the
     // strides.
     template<typename Ptr, typename ...Ptrs>
+    HALIDE_ALWAYS_INLINE
```

Comment:
Similarly, several of these helpers could be moved into a hypothetical `struct BufferHelper` to eliminate extra instantiations. (Again, not sure if it pays for itself... just sayin'.)